packages/connect/src/protocol-grpc/validate-trailer.ts

```diff
@@ -28,4 +31,12 @@ export function validateTrailer(trailer: Headers, header: Headers): void {
     });
     throw err;
   }
+  if (
+    !header.has(headerGrpcStatus) &&
+    !header.has(headerStatusDetailsBin) &&
+    !trailer.has(headerGrpcStatus) &&
+    !trailer.has(headerStatusDetailsBin)
+  ) {
+    throw new ConnectError("protocol error: missing status", Code.Internal);
```

Comment:
`validateResponse()` returns a `foundStatus: boolean` that only looks for the `grpc-status` field. I don't have a suggestion yet, but we should try to clean this up a bit.

Reply:
Discussed offline: We'll look into cleaning this up in v2.